registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/StageHealthCheckHandler.java

```diff
@@ -144,39 +147,48 @@ public StageHealthCheckHandler register(String name, long timeout, Handler<Promi
-		try {
+	public void queueHealthChecker(Promise<Status> promise, MosipQueueManager<MosipQueue, byte[]> mosipQueueManager, MosipQueueConnectionFactory<MosipQueue> mosipConnectionFactory) {
 
-			String message = "Ping";
-			MosipQueue mosipQueue = new MosipActiveMq(HealthConstant.QUEUE_ADDRESS, queueUsername, queuePassword, queueBrokerUrl);
-			mosipQueueManager.send(mosipQueue, message.getBytes(), HealthConstant.QUEUE_ADDRESS);
-
-			ActiveMQConnectionFactory activeMQConnectionFactory = new ActiveMQConnectionFactory(queueUsername,
-					queuePassword, queueBrokerUrl);
-			ActiveMQConnection activemQConn = (ActiveMQConnection) connection;
-			if (activemQConn == null || activemQConn.isClosed()) {
-				connection = activeMQConnectionFactory.createConnection();
-				activemQConn = (ActiveMQConnection) connection;
-				activemQConn.addTransportListener(new TransportExceptionListener());
-				if (session == null) {
-					connection.start();
-					this.session = this.connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
-					Queue destination = session.createQueue(HealthConstant.QUEUE_ADDRESS);
-					messageConsumer = session.createConsumer(destination);
-				}
-			}
-			String res = new String(((ActiveMQBytesMessage) messageConsumer.receive()).getContent().data);
-			if (res == null || !message.equalsIgnoreCase(res)) {
-				final JsonObject result = resultBuilder.create().add(HealthConstant.ERROR, "Could not read response from queue").build();
-				promise.complete(Status.KO(result));
+		try {
+			final String msg = "Ping";
+
+			if (mosipQueue == null)
+				mosipQueue = mosipConnectionFactory.createConnection("ACTIVEMQ", queueUsername,
+						queuePassword, queueBrokerUrl);
+
+			mosipQueueManager.send(mosipQueue, msg.getBytes(), HealthConstant.QUEUE_ADDRESS);
+
+			if (!isConsumerStarted) {
+				QueueListener listener = new QueueListener() {
+					@Override
+					public void setListener(Message message) {
+						try {
+							consumerListener(message, promise, msg);
+						} catch (Exception e) {
+							LOGGER.error(LoggerFileConstant.SESSIONID.toString(),
+									LoggerFileConstant.REGISTRATIONID.toString(), "Unable to check activemq health", ExceptionUtils.getStackTrace(e));
+						}
+					}
+				};
+				mosipQueueManager.consume(mosipQueue, HealthConstant.QUEUE_ADDRESS, listener);
+				isConsumerStarted = true;
 			}
-			final JsonObject result = resultBuilder.create().add(HealthConstant.RESPONSE, res).build();
-			promise.complete(Status.OK(result));
 		} catch (Exception e) {
+			isConsumerStarted = false;
+			mosipQueue = null;
 			final JsonObject result = resultBuilder.create().add(HealthConstant.ERROR, e.getMessage()).build();
 			promise.complete(Status.KO(result));
 		}
 	}
+	public void consumerListener(Message message, Promise<Status> promise, String msg) {
+		String res = new String(((ActiveMQBytesMessage) message).getContent().data);
+		if (res == null || !msg.equalsIgnoreCase(res)) {
```

Comment:
How can "res" become null? We are just initialising. Is it a typo? Are you trying to chk the msg?